engine/src/test/java/org/camunda/bpm/engine/test/api/task/TaskServiceTest.java

```diff
@@ -2219,35 +2197,68 @@ public void testDeleteAttachment() throws ParseException {
   @Test
   public void testDeleteTaskAttachmentWithNullParameters() {
     int historyLevel = processEngineConfiguration.getHistoryLevel().getId();
-    if (historyLevel> ProcessEngineConfigurationImpl.HISTORYLEVEL_NONE) {
+    if (historyLevel > ProcessEngineConfigurationImpl.HISTORYLEVEL_NONE) {
       try {
         taskService.deleteTaskAttachment(null, null);
         fail("expected process engine exception");
-      } catch (ProcessEngineException e) {}
+      } catch (ProcessEngineException e) {
+      }
     }
   }
 
   @Test
-  public void testDeleteTaskAttachmentWithTaskIdNull() {
-    int historyLevel = processEngineConfiguration.getHistoryLevel().getId();
-    if (historyLevel> ProcessEngineConfigurationImpl.HISTORYLEVEL_NONE) {
-      try {
-        taskService.deleteTaskAttachment(null, "myAttachmentId");
-        fail("expected process engine exception");
-      } catch(ProcessEngineException e) {}
-    }
+  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_NONE)
+  public void testDeleteTaskAttachmentThatDoesNotExist() {
+    assertThatThrownBy(() -> taskService.deleteTaskAttachment(null, "attachmentDoesNotExist")).isInstanceOf(
+            NullValueException.class)
+        .hasMessageContaining("No attachment exists with attachmentId 'attachmentDoesNotExist'");
+  }
+
+  @Test
+  @Deployment(resources = { "org/camunda/bpm/engine/test/api/oneTaskProcess.bpmn20.xml" })
+  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_NONE)
+  public void testDeleteTaskAttachmentWithTaskIdEmpty() {
+    // given
+    runtimeService.startProcessInstanceByKey("oneTaskProcess");
+    Attachment attachment = taskService.createAttachment("web page", "", null, "weatherforcast",
+        "temperatures and more", new ByteArrayInputStream("someContent".getBytes()));
```

Comment:
Based on my understanding of the issue, I think it would also make sense to look into prohibiting the creation of task attachments with empty taskId and null processInstanceId. At the moment, this is still allowed because the creation only [checks](https://github.com/camunda/camunda-bpm-platform/blob/119cbf8804b79fd53310ed42ebd567f6e26812fa/engine/src/main/java/org/camunda/bpm/engine/impl/cmd/CreateAttachmentCmd.java#L71) for the taskId not being null. I can open a follow up ticket you agree that this makes sense